vms/platformvm/txs/mempool/mempool.go

```diff
@@ -65,10 +116,87 @@ func (m *mempool) Add(tx *txs.Tx) error {
 	default:
 	}
 
-	return m.Mempool.Add(tx)
+	complexity, gas, err := m.meterer.Meter(tx.Unsigned)
+	if err != nil {
+		return err
+	}
+
+	if err := m.mempool.Add(tx); err != nil {
```

Comment:
I think part of the goal of this (series of?) PR was to replace the MaxTxSize limit with a gas limit check. If we do that, then we can remove the hack here: https://github.com/ava-labs/avalanchego/blob/master/vms/platformvm/block/executor/manager.go#L172-L188

Reply:
I think it might make sense to do this as a separate follow-up, I was planning on scoping this PR to just change the mempool to be sorted. I think the follow-ups would be to update this check + another one to update the block builder to do a gas price check.

Reply:
Follow-up: https://github.com/ava-labs/avalanchego/issues/4287